09-valve-monitor/firmware/zephyr/src/main.c

```diff
@@ -674,54 +715,52 @@ void main(void)
+            // Process all pending inbound requests.
             while (true) {
-                /* Pop the next available note from data.qi. */
+                // Pop the next available note from data.qi.
                 J *req = NoteNewRequest("note.get");
                 JAddStringToObject(req, "file", "data.qi");
                 JAddBoolToObject(req, "delete", true);
                 J *rsp = NoteRequestResponse(req);
                 if (rsp != NULL) {
 
-                    /*
-                     * If an error is returned, this means that no response is
-                     * pending. Note that it's expected that this might return
-                     * either a "note does not exist" error if there are no
-                     * pending inbound notes, or a "file does not exist" error
-                     * if the inbound queue hasn't yet been created on the
-                     * service.
-                     */
+                    // If an error is returned, this means that no response is
+                    // pending. Note that it's expected that this might return
+                    // either a "note does not exist" error if there are no
+                    // pending inbound notes, or a "file does not exist" error
+                    // if the inbound queue hasn't yet been created on the
+                    // service.
                     if (NoteResponseError(rsp)) {
                         NoteDeleteResponse(rsp);
                         break;
                     }
 
-                    /* Get the note's body. */
+                    // Get the note's body.
                     J *body = JGetObject(rsp, "body");
                     if (body != NULL) {
                         char *cmd = JGetString(body, "state");
                         if (cmd != NULL && strlen(cmd) != 0) {
                             handleValveCmd(cmd);
-                            /*
-                             * If we received a valve command (open or close),
-                             * we want to publish the system status immediately
-                             * in response, regardless of if it's time to do so
-                             * based on the monitor interval. This acts as a
-                             * sort of acknowledgment so that the controller 
-                             * knows their valve command was received.
-                             */
+                            // If we received a valve command (open or close),
+                            // we want to publish the system status immediately
+                            // in response, regardless of if it's time to do so
+                            // based on the monitor interval. This acts as a
+                            // sort of acknowledgment so that the controller
+                            // knows their valve command was received.
+                            state.publishRequired = true;
```

Comment:
Thinking about this more after our conversation, we could potentially simplify things by always setting `publishRequired` to true in `handleValveCmd`. That way, even if we don't toggle (because, e.g., valve is closed and user requested close), we still publish. Not strictly necessary to change, though. Just thought I'd mention it.

Reply:
I had considered it too, but was too lazy. This was the push I needed. :smile: